tpe/plugin/src/SimulationFeatures_TEST.cc

```diff
@@ -0,0 +1,149 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/eigen3/Conversions.hh>
+
+#include <ignition/physics/FindFeatures.hh>
+#include <ignition/plugin/Loader.hh>
+#include <ignition/physics/RequestEngine.hh>
+
+// Features
+#include <ignition/physics/ForwardStep.hh>
+#include <ignition/physics/GetEntities.hh>
+#include <ignition/physics/Shape.hh>
+#include <ignition/physics/sdf/ConstructWorld.hh>
+
+#include <sdf/Root.hh>
+#include <sdf/World.hh>
+
+#include <test/PhysicsPluginsList.hh>
+#include <test/Utils.hh>
+
+struct TestFeatureList : ignition::physics::FeatureList<
+  ignition::physics::ForwardStep,
+  ignition::physics::GetEngineInfo,
+  ignition::physics::GetWorldFromEngine,
+  ignition::physics::GetModelFromWorld,
+  ignition::physics::GetLinkFromModel,
+  ignition::physics::GetShapeFromLink,
+  ignition::physics::GetShapeBoundingBox,
+  ignition::physics::sdf::ConstructSdfWorld
+> { };
+
+using TestWorldPtr = ignition::physics::World3dPtr<TestFeatureList>;
+using TestShapePtr = ignition::physics::Shape3dPtr<TestFeatureList>;
+
+std::unordered_set<TestWorldPtr> LoadWorlds(
+    const std::string &_library,
+    const std::string &_world)
+{
+  ignition::plugin::Loader loader;
+  loader.LoadLib(_library);
+
+  const std::set<std::string> pluginNames =
+    ignition::physics::FindFeatures3d<TestFeatureList>::From(loader);
+
+  EXPECT_LT(0u, pluginNames.size());
+
+  std::unordered_set<TestWorldPtr> worlds;
+  for (const std::string &name : pluginNames)
+  {
+    ignition::plugin::PluginPtr plugin = loader.Instantiate(name);
+
+    std::cout << " -- Plugin name: " << name << std::endl;
+
+    auto engine =
+      ignition::physics::RequestEngine3d<TestFeatureList>::From(plugin);
+    EXPECT_NE(nullptr, engine);
+    sdf::Root root;
+    const sdf::Errors &errors = root.Load(_world);
+    EXPECT_EQ(0u, errors.size());
+    const sdf::World *sdfWorld = root.WorldByIndex(0);
+    auto world = engine->ConstructWorld(*sdfWorld);
+
+    worlds.insert(world);
+  }
+  return worlds;
+}
+
+class SimulationFeatures_TEST
+  : public ::testing::Test,
+    public ::testing::WithParamInterface<std::string>
+{};
+
+TEST_P(SimulationFeatures_TEST, ShapeBoundingBox)
+{
+  const std::string library = GetParam();
+  if (library.empty())
+    return;
+
+  auto worlds = LoadWorlds(library, TEST_WORLD_DIR "/falling.world");
```

Comment:
I think this world needs to be copied over from `dartsim`.

Or even better, moved to a common location so both plugins can use the same files in their tests.